sobczak_julia_lidar_uncertainty.py

```diff
@@ -0,0 +1,206 @@
+# Set SOAP plot locations path
+soap_plots_path = os.path.join(soap_dir,
+                        'vector_data',
+                         'SOAP_centroids.shp')
+# Set SOAP LiDAR Canopy Height Model path
+soap_chm_path = os.path.join(soap_dir,
+                         '2013',
+                          'lidar',
+                         'SOAP_lidarCHM.tif')
+# Check if SOAP paths exist
+path_check_2 = [soap_dir, soap_insitu_path, soap_plots_path, soap_chm_path]
+for path in path_check_2:
+    print(os.path.exists(path))
+
+
+# ### Step 3: Start downloading data from earthpy 
+# 
+# Find original data and activity in the [Earth Data Analytics textbook](https://www.earthdatascience.org/courses/use-data-open-source-python/spatial-data-applications/lidar-remote-sensing-uncertainty/summarize-and-compare-lidar-insitu-tree-height/).
+# 
+
+# In[3]:
+
+
+# Use earthpy to import data
+et.data.get_data('spatial-vector-lidar')
+
+
+# ### Step 4: Calculate zonal stats
+# Use the zstats_meanmax() function and calculate for both NEON sites. View the function in clean.py directory.
+
+# In[4]:
+
+
+# SJER zstats
+sjer_zstats = clean.zstats_meanmax(sjer_plots_path, sjer_chm_path)
+# SOAP zstats
+soap_zstats = clean.zstats_meanmax(soap_plots_path, soap_chm_path)
+
+# Check datasets
+soap_zstats.head()
+
+
+# ### Step 5:  Get stats for insitu data
+# Use the insitu_meanmax() function and calculate for both NEON sites. View the function in clean.py directory.
+
+# In[5]:
+
+
+# SJER site insitu data
+sjer_insitu = clean.insitu_meanmax(sjer_insitu_path, 'plotid')
+# SOAP site insitu data
+soap_insitu = clean.insitu_meanmax(soap_insitu_path, 'plotid')
+
+# Check datasets
+sjer_insitu.head(), soap_insitu.head()
+
+
+# ### Step 6: Merge datasets
+# Use the merge_df() function and calculate for both NEON sites. View the function in clean.py directory.
+
+# In[6]:
+
+
+# Adjust SOAP site ID's before merging
+if not soap_zstats['ID'].str.contains('SOAP').any():
+    soap_zstats['ID'] = ('SOAP' + soap_zstats['ID'])
+    
+# Merge data for SOAP site
+soap_merge_df = clean.merge_df(soap_zstats, soap_insitu, id_col='ID')
+
+# Merge data for SJER site
+sjer_merge_df = clean.merge_df(sjer_zstats, sjer_insitu, id_col='Plot_ID')
+
+sjer_merge_df.head(), soap_merge_df.head()
+
+
+# ### Step 7: Plot the data
+# 
+# The regression plots in seaborn are primarily intended to add a visual guide that helps to emphasize patterns in a dataset during exploratory data analyses. Regression plots as the name suggests creates a regression line between 2 parameters and helps to visualize their linear relationships ([source: geeksforgeeks.org](https://www.geeksforgeeks.org/seaborn-regression-plots/)).
+
+# In[7]:
+
+
+# Run plotting function over NEON sites
+sjer_plot = clean.plot_mergedf('San Joaquin Experimental Range', sjer_merge_df)
+soap_plot = clean.plot_mergedf('Soaproot Saddle', soap_merge_df)
+
+sjer_plot, soap_plot
+
+
+# ### Alternate option: use the class saved in clean.py
```

Comment:
This last section could be written out a little better. Perhaps also showcase a plot or the df head to prove this alternative works?